x/cdp/abci_test.go

```diff
@@ -174,6 +221,74 @@ func (suite *ModuleTestSuite) TestSeizeSingleCdpWithFees() {
+	// test setup, creating
+	// 50 xrp cdps each with
+	// collateral: 10000000000
+	// debt: between 750000000 - 1249000000
+	// if debt above 10000000000,
+	//     cdp added to tracker / liquidation list
+	//     debt total added to trackers debt total
+	// 50 btc cdps each with
+	// collateral: 10000000000
+	// debt: between 2700000000 - 5332000000
+	// if debt above 4000000000,
+	//     cdp added to tracker / liquidation list
+	//     debt total added to trackers debt total
+
+	// naively we expect roughly half of the cdps to be above the debt tracking floor, roughly 25 of them collaterallized with xrp, the other 25 with btcb
+
+	// usdx is the principal for all cdps
+	suite.createCdps()
+	ak := suite.app.GetAccountKeeper()
+	bk := suite.app.GetBankKeeper()
+
+	// set the cdp begin blocker to run every other block
+	params := suite.keeper.GetParams(suite.ctx)
+	params.BeginBlockerExecutionBlockInterval = 2
+	suite.keeper.SetParams(suite.ctx, params)
+
+	// test case 1 setup
+	acc := ak.GetModuleAccount(suite.ctx, types.ModuleName)
+	// track how much xrp collateral exists in the cdp module
+	originalXrpCollateral := bk.GetBalance(suite.ctx, acc.GetAddress(), "xrp").Amount
+	// set the trading price for xrp:usd pools
+	suite.setPrice(d("0.2"), "xrp:usd")
+
+	// test case 1 execution
+	cdp.BeginBlocker(suite.ctx, abci.RequestBeginBlock{Header: suite.ctx.BlockHeader()}, suite.keeper)
+
+	// test case 1 assert
+	acc = ak.GetModuleAccount(suite.ctx, types.ModuleName)
+	// get the current amount of xrp held by the cdp module
+	finalXrpCollateral := bk.GetBalance(suite.ctx, acc.GetAddress(), "xrp").Amount
+	seizedXrpCollateral := originalXrpCollateral.Sub(finalXrpCollateral)
+	// calculate the number of cdps that were liquidated based on the total
+	// seized collateral divided by the size of each cdp when it was created
+	xrpLiquidations := int(seizedXrpCollateral.Quo(i(10000000000)).Int64())
+	// should be 0 because the cdp begin blocker is configured to
+	// skip execution every odd numbered block
+	suite.Equal(0, xrpLiquidations)
+
+	// test case 2 setup
+	// simulate running the second block of the chain
+	suite.ctx = suite.ctx.WithBlockHeight(2)
+
+	// test case 2 execution
+	cdp.BeginBlocker(suite.ctx, abci.RequestBeginBlock{Header: suite.ctx.BlockHeader()}, suite.keeper)
+
+	// test case 2 assert
+	acc = ak.GetModuleAccount(suite.ctx, types.ModuleName)
+	// get the current amount of xrp held by the cdp module
+	finalXrpCollateral = bk.GetBalance(suite.ctx, acc.GetAddress(), "xrp").Amount
+	seizedXrpCollateral = originalXrpCollateral.Sub(finalXrpCollateral)
+	// calculate the number of cdps that were liquidated based on the total
+	// seized collateral divided by the size of each cdp when it was created
+	xrpLiquidations = int(seizedXrpCollateral.Quo(i(10000000000)).Int64())
+	// should be 10 because of magic numbers elsewhere in this file
+	suite.Equal(10, xrpLiquidations)
```

Comment:
ditto on message here since we are focused on running vs not running and not the number in the test.  Could assert length greater than zero here as well to remove dependence on numbers since that's not the goal of the test